src/web/components/ApiKeyManagement/KeyCreationDialog.tsx

```diff
@@ -0,0 +1,181 @@
+  ];
+
+  const [uncopiedValueNames, setUncopiedValueNames] = useState(
+    secrets.map((secret) => secret.valueName)
+  );
+
+  const onCopyGenerator = (copiedValueName: String) => {
+    return () => {
+      setUncopiedValueNames((oldUncopiedValueNames: string[]) => {
+        return oldUncopiedValueNames.filter((valueName) => valueName !== copiedValueName);
+      });
+    };
+  };
+
+  const onClose = () => {
+    if (uncopiedValueNames.length > 0) {
+      showPopupMessage('Please copy all secrets shown before closing the page');
+      return;
+    }
+
+    closeDialog();
+  };
+
+  return (
+    <div>
+      <h1>{keySecrets.name} Secrets</h1>
+      <p>
+        Please copy the key and secret as they will not be saved after this window is closed. Keep
+        these secrets in a secure location and do not share them with anyone. If the secrets are
+        lost a new key will have to be generated.
+      </p>
+      {secrets.map((secret) => (
+        <div key={secret.valueName}>
+          <h2>{secret.valueName}</h2>
+          <DisplaySecret secret={secret} onCopy={onCopyGenerator(secret.valueName)} />
+        </div>
+      ))}
+      <div className='cancel-container'>
+        <button type='button' className='transparent-button' onClick={onClose}>
+          Close
+        </button>
+      </div>
+    </div>
+  );
+}
+
+function KeyCreationDialog({
+  onKeyCreation,
+  triggerButton,
+  availableRoles,
+}: KeyCreationDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [keySecrets, setKeySecrets] = useState<KeySecretProp>(undefined);
+  const [showStatusPopup, setShowStatusPopup] = useState<boolean>(false);
+  const [statusPopupMessage, setStatusPopupMessage] = useState<string>('');
+
+  const showPopupMessage = (message: string) => {
+    setStatusPopupMessage(message);
+    setShowStatusPopup(true);
+  };
+
+  const onFormSubmit: SubmitHandler<CreateApiKeyFormDTO> = async (formData) => {
+    setKeySecrets(await onKeyCreation(formData));
+    showPopupMessage(
+      'Your key has been created. Please copy the credentials before closing the window.'
+    );
+  };
+
+  const closeDialog = () => {
+    setKeySecrets(undefined);
+    setShowStatusPopup(false);
+    setOpen(false);
+  };
+
+  return (
+    <div className='key-creation-dialog'>
+      <Dialog triggerButton={triggerButton} open={open} onOpenChange={setOpen} hideClose>
+        {!keySecrets ? (
+          <CreateApiKeyForm
+            onFormSubmit={onFormSubmit}
+            availableRoles={availableRoles}
+            closeDialog={closeDialog}
+          />
+        ) : (
+          <ShowApiKeySecrets
+            closeDialog={closeDialog}
+            keySecrets={keySecrets}
+            showPopupMessage={showPopupMessage}
+          />
+        )}
+
+        {showStatusPopup && (
+          <StatusPopup
+            status='Info'
+            show={showStatusPopup}
+            setShow={setShowStatusPopup}
+            message={statusPopupMessage}
```

Comment:
this might deserve a longer display duration e.g. `displayDuration={10000}` since it's possibly not super clear to users what they need to do initially.

Reply:
Yeah that makes sense